modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -312,54 +325,73 @@ namespace {{packageName}}.Client
+            if (type == typeof(Stream)) {
+                if (headers != null) {
                     var filePath = String.IsNullOrEmpty(Configuration.TempFolderPath)
                         ? Path.GetTempPath()
                         : Configuration.TempFolderPath;
                     var regex = new Regex(@"Content-Disposition=.*filename=['""]?([^'""\s]+)['""]?$");
-                    foreach (var header in headers)
-                    {
+                    foreach (var header in headers) {
                         var match = regex.Match(header.ToString());
-                        if (match.Success)
-                        {
+                        if (match.Success) {
                             string fileName = filePath + SanitizeFilename(match.Groups[1].Value.Replace("\"", "").Replace("'", ""));
                             File.WriteAllBytes(fileName, response.RawBytes);
                             return new FileStream(fileName, FileMode.Open);
                         }
                     }
                 }
                 var stream = new MemoryStream(response.RawBytes);
+                log.Trace("end");
                 return stream;
             }
 
-            if (type.Name.StartsWith("System.Nullable`1[[System.DateTime")) // return a datetime object
-            {
+            if (type.Name.StartsWith("System.Nullable`1[[System.DateTime")) {
+                // return a datetime object 
+                log.Trace("end");
                 return DateTime.Parse(response.Content,  null, System.Globalization.DateTimeStyles.RoundtripKind);
             }
 
-            if (type == typeof(String) || type.Name.StartsWith("System.Nullable")) // return primitive type
-            {
+            if (type == typeof(String) || type.Name.StartsWith("System.Nullable")) {
+                // return primitive type 
+                log.Trace("end");
                 return ConvertType(response.Content, type);
             }
 
+            log.Trace("end");
+            return deserialize(response, type);
+        }
+
+        /// <summary>
+        /// Deserialize the JSON string into a proper object.
+        /// </summary>
+        /// <typeparam name="T">Model object</typeparam>
+        /// <param name="response">The HTTP response.</param>
+        /// <returns>Object representation of the JSON string.</returns>
+        public T Deserialize<T>(IRestResponse response) {
+            log.Trace("start");
+            var type = typeof(T);
             // at this point, it must be a model (json)
-            try
-            {
-                return JsonConvert.DeserializeObject(response.Content, type, serializerSettings);
+            if(!(deserialize(response, type) is T)) {
+                return default(T);
             }
-            catch (Exception e)
-            {
+
+            return (T) deserialize(response, type);
+        }
+
+        private object deserialize(IRestResponse response,Type type) {
```

Comment:
`deserialize` should be capitalized to follow C# coding standards. Since this would conflict with the existing `Deserialize` method, I'd recommend calling it `DeserializeObject` to match the `JsonConvert` method being abstracted.